jenkins/deploy-clio-server.sh

```diff
@@ -0,0 +1,209 @@
+fi
+
+# build the clio-server docker image
+docker run --rm \
+    -v /var/run/docker.sock:/var/run/docker.sock \
+    -v ${CLIO_DIR}:/clio \
+    -w="/clio" \
+    broadinstitute/scala:scala-2.11.8 sbt "$SET_TESTS" "$SET_VERSION" "clio-server/docker"
+
+# if no tag was given, read the default tag from the generated resource file
+if [[ -z "$DOCKER_TAG" ]]; then
+    DOCKER_TAG=$(cat ${CLIO_DIR}/clio-server/target/scala-2.12/resource_managed/main/clio-server-version.conf | perl -pe 's/clio\.server\.version:\s+//g')
+fi
+
+# push the docker image
+docker push broadinstitute/clio-server:${DOCKER_TAG}
+
+# TMPDIR
+TMPDIR=$(mktemp -d ${PROG_NAME}-XXXXXX)
+# temporary env file for rendering ctmpls
+CTMPL_ENV_FILE="${TMPDIR}/env-vars.txt"
+
+# populate temp env.txt file with necessary environment
+echo "ENV=${ENV}" >> ${CTMPL_ENV_FILE}
+echo "DOCKER_TAG=${DOCKER_TAG}" >> ${CTMPL_ENV_FILE}
+echo "APP_DIR=${APP_DIR}" >> ${CTMPL_ENV_FILE}
+echo "CLIO_CONF_DIR=${CLIO_CONF_DIR}" >> ${CTMPL_ENV_FILE}
+echo "CLIO_LOG_DIR=${CLIO_LOG_DIR}" >> ${CTMPL_ENV_FILE}
+echo "CLIO_APP_CONF=${CLIO_APP_CONF}" >> ${CTMPL_ENV_FILE}
+echo "CLIO_LOGBACK_CONF=${CLIO_LOGBACK_CONF}" >> ${CTMPL_ENV_FILE}
+echo "CLIO_ENV_FILE=${CLIO_ENV_FILE}" >> ${CTMPL_ENV_FILE}
+echo "HOST_CLIO_PORT=${HOST_CLIO_PORT}" >> ${CTMPL_ENV_FILE}
+echo "CONTAINER_CLIO_PORT=${CONTAINER_CLIO_PORT}" >> ${CTMPL_ENV_FILE}
+
+# copy configs to temp dir
+cp ${CONFIG_DIR}/* ${TMPDIR}/
+
+# render all ctmpls
+docker run --rm \
+    -v ${PWD}/${TMPDIR}:/working \
+    -v ${PWD}/jenkins:/scripts \
+    -v ${TOKEN_FILE}:/root/.vault-token:ro \
+    --env-file="${CTMPL_ENV_FILE}" \
+    broadinstitute/dsde-toolbox:latest /scripts/render-ctmpl.sh
+rm -f ${CTMPL_ENV_FILE}
+
+stopcontainer() {
+    # stop running container on host
+    ${SSHCMD} ${SSH_USER}@${CLIO_HOST} "test -f ${APP_DIR}/docker-compose.yml && docker-compose -p clio-server -f ${APP_DIR}/docker-compose.yml stop || echo"
+
+    # docker-compose rm -f
+    ${SSHCMD} ${SSH_USER}@${CLIO_HOST} "test -f ${APP_DIR}/docker-compose.yml && docker-compose -p clio-server -f ${APP_DIR}/docker-compose.yml rm -f || echo"
+}
+
+startcontainer() {
+    # docker-compose pull
+    ${SSHCMD} ${SSH_USER}@${CLIO_HOST} "docker-compose -p clio-server -f ${APP_DIR}/docker-compose.yml pull"
+
+    # docker-compose up -d
+    ${SSHCMD} ${SSH_USER}@${CLIO_HOST} "docker-compose -p clio-server -f ${APP_DIR}/docker-compose.yml up -d"
+}
+
+# copy configs to temp directory on host
+# rm -rf /app.new
+# mkdir /app.new
+${SSHCMD} ${SSH_USER}@${CLIO_HOST} "sudo rm -rf ${APP_DIR}.new && sudo mkdir ${APP_DIR}.new && sudo chgrp ${SSH_USER} ${APP_DIR}.new && sudo chmod g+w ${APP_DIR}.new"
+
+# copy configs to /app.new
+${SCPCMD} -r ${TMPDIR}/* ${SSH_USER}@${CLIO_HOST}:${APP_DIR}.new/
+
+# stop running the existing container
+stopcontainer
+
+# move /app.old to /app.old.old
+# move /app to /app.old
+# move /app.new to /app
+${SSHCMD} ${SSH_USER}@${CLIO_HOST} <<EOF
```

Comment:
I only keep the `.old.old` so we can roll back the directories if the deploy fails. Once we verify it succeeded, we delete it.